### ISSUE TITLE
`tinyglobby` can hang on invalid `fast-glob` backslash usage

### ISSUE THREAD

**Issue by userquin** [SuperchupuDev/tinyglobby#72]
Check https://github.com/vuejs/vitepress/issues/4357

**Comment by SuperchupuDev** [comment]
i'm on windows and absolute paths should work, can you provide a minimal repro? whatever it is #69 will probably solve it

**Comment by userquin** [comment]
clone VP repo then `pnpm install && pnpm build`

**Comment by userquin** [comment]
https://github.com/mrmlnc/fast-glob?tab=readme-ov-file#how-to-write-patterns-on-windows

fast-glob handles that, but doesn't recommend

Using simple replacement works:  

![imagen](https://github.com/user-attachments/assets/8d0cdb89-3157-4a72-8e77-453420899a92)

**Comment by userquin** [comment]
@SuperchupuDev https://github.com/mrmlnc/fast-glob/blob/da648078ae87bce81fcd42e64d5b2b1360c47b09/src/utils/path.ts#L57

**Comment by SuperchupuDev** [comment]
will try to make it as consistent to fast-glob as possible

**Comment by Torathion** [comment]
From what I've gathered: The windows glob issues are very important. OP came from vitepress having this issue and remix and analog also referring to this issue. I wanted to make a PR in npm-check-updates and came across the same issue as a dozen tests started to fail after switching from fast-glob to tinyglobby.

**Comment by Sytten** [comment]
I would also say yes this is an issue, we use tsup to build plugins in pr ecosystem and  got report of process hanging on windows. Ot does look like it is related to tinyglobby.

**Comment by SuperchupuDev** [comment]
sorry to keep y'all waiting, the next release is currently being blocked by some perf implementation i've been trying to figure out for a while. once that's solved i'll fix this and release a new version

**Comment by SuperchupuDev** [comment]
i'm working on this right now, but i can't seem to find a single pattern with `\` as separators that make fast-glob work. @Torathion @userquin can y'all share a minimal reproduction of a case in which fast-glob returns results correctly like that? to then change tinyglobby so that it matches `fast-glob`'s functionality

**Comment by Torathion** [comment]
Unfortunately I can't help you with this as I'm not very familiar with globs. I only found out about it when the tests in NCU suddenly failed after using it as a drop-in replacement.

But I see that you only have tests using `/`, while you don't have any tests using `\\` alone. Maybe adding these tests would help to establish `tinyglobby` as drop-in replacement for `fast-glob`.

**Comment by userquin** [comment]
https://github.com/userquin/tinyglobby-issue-72

**Comment by SuperchupuDev** [comment]
@userquin `fg.mjs` doesn't seem to return anything in your repro. is there any case in which fast-glob returns something with windows paths as likely expected, or is the issue that tinyglobby doesn't return `[]`? i thought the expected thing was support for those patterns

![Image](https://github.com/user-attachments/assets/f47d3a6b-294d-4194-828e-75145a4ed36c)

**Comment by userquin** [comment]
Yeah, `fg.mjs` doesn't hang while `tg.mjs` does (the original issue)

https://github.com/mrmlnc/fast-glob?tab=readme-ov-file#how-to-write-patterns-on-windows

**Comment by SuperchupuDev** [comment]
okay, so as far as i understand the problem you all are having is that that invalid fast-glob usage hangs, but it's invalid fg usage regardless. is that right?

**Comment by SuperchupuDev** [comment]
in theory the new optimizer i merged a few days ago should make this not hang anymore, but it still creates an odd internal state regardless

**Comment by SuperchupuDev** [comment]
@userquin tinyglobby `0.2.11` is out which should fix the perf problems thanks to new optimizations introduced. can you tell me if it still hangs? locally reproducing it doesn't seem to hang anymore, at least on a friend's computer

**Comment by userquin** [comment]
@SuperchupuDev now it works:

![Image](https://github.com/user-attachments/assets/b27cec0a-0070-43fa-a7e4-e6b2781e86d4)
